webpack.config.extensions.js

```diff
@@ -36,112 +28,50 @@ function blockScripts( type, inputDir, presetBlocks ) {
+		viewBlocks[ block + '/view' ] = [ viewSetup, ...[ viewScriptPath ] ];
+	}
+	return viewBlocks;
+}, {} );
 
-	// Combines all the different blocks into one editor-beta.js script
-	const editorBetaScript = [
-		editorSetup,
-		...blockScripts( 'editor', path.join( __dirname, 'extensions' ), allPresetBlocks ),
-	];
+// Combines all the different blocks into one editor.js script
+const editorScript = [
+	editorSetup,
+	...blockScripts( 'editor', path.join( __dirname, 'extensions' ), presetBlocks ),
+];
 
-	const webpackConfig = {
-		bail: ! isDevelopment,
-		context: __dirname,
-		entry: {
-			editor: editorScript,
-			'editor-beta': editorBetaScript,
-			...viewBlocksScripts,
-		},
-		mode: isDevelopment ? 'development' : 'production',
-		devtool: process.env.SOURCEMAP || ( isDevelopment ? '#eval' : false ),
-		output: {
-			path: path.join( __dirname, '_inc', 'blocks' ),
-			filename: '[name].js',
-			libraryTarget: 'window',
-		},
-		optimization: {
-			minimize: ! isDevelopment,
-			minimizer: Minify( {
-				cache: process.env.CIRCLECI
-					? `${ process.env.HOME }/terser-cache`
-					: 'docker' !== process.env.CONTAINER,
-				parallel: workerCount,
-				sourceMap: Boolean( process.env.SOURCEMAP ),
-				terserOptions: {
-					ecma: 5,
-					safari10: true,
-					mangle: true,
-				},
-			} ),
-		},
-		module: {
-			rules: [
-				TranspileConfig.loader( {
-					workerCount,
-					configFile: path.join( __dirname, 'babel.config.js' ),
-					cacheDirectory: true,
-					exclude: /node_modules\//,
-				} ),
-				SassConfig.loader( {
-					preserveCssCustomProperties: false,
-					includePaths: [ path.join( __dirname, 'client' ) ],
-					prelude: '@import "~@automattic/calypso-color-schemes/src/shared/colors";',
-				} ),
-				FileConfig.loader(),
-			],
-		},
-		resolve: {
-			extensions: [ '.json', '.js', '.jsx' ],
-			modules: [ 'node_modules' ],
-		},
-		node: false,
-		plugins: [
-			new webpack.DefinePlugin( {
-				'process.env.NODE_ENV': JSON.stringify( process.env.NODE_ENV ),
-				global: 'window',
-			} ),
-			new webpack.IgnorePlugin( /^\.\/locale$/, /moment$/ ),
-			...SassConfig.plugins( { cssFilename, minify: ! isDevelopment } ),
-			new DuplicatePackageCheckerPlugin(),
-			new CopyWebpackPlugin( [
-				{
-					from: presetPath,
-					to: 'index.json',
-				},
-			] ),
-		],
-		externals: [ wordpressExternals, 'wp', 'lodash' ],
-	};
+// Combines all the different blocks into one editor-beta.js script
+const editorBetaScript = [
+	editorSetup,
+	...blockScripts( 'editor', path.join( __dirname, 'extensions' ), allPresetBlocks ),
+];
 
-	return webpackConfig;
-}
+const webpackConfig = getBaseWebpackConfig( null, {
+	entry: {
+		editor: editorScript,
+		'editor-beta': editorBetaScript,
+		...viewBlocksScripts,
+	},
+	'output-path': path.join( __dirname, '_inc', 'blocks' ),
+} );
 
-module.exports = getWebpackConfig;
+module.exports = _.merge( {}, webpackConfig, {
+	plugins: [
+		new CopyWebpackPlugin( [
```

Comment:
I expected not needing to copy this file anymore? Re https://github.com/Automattic/jetpack/pull/11801

Reply:
Jon and I discussed this at length during our call yesterday. While we still want to remove it eventually, getting it right is complex, and I definitely wanted to unblock this PR -- so keeping it for the time being made the most sense :slightly_smiling_face:

Reply:
Makes sense 👍  I just wasn't fully aware at which order things are happening, as I remembered seeing a PR about it in the past. :-)